packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryResetPassword/index.tsx

```diff
@@ -64,82 +112,236 @@ const AccountRecoveryResetPassword = ({
-          <LinkRememberPassword {...{ email, forceAuth }} />
-        </>
+        </Banner>
+      )}
+      {BannerState.PasswordResendError === bannerState && (
+        <Banner type={BannerType.error}>
+          <FtlMsg id="account-recovery-reset-password-email-resend-error">
+            <p>
+              Sorry, there was a problem resending a reset password link to your
+              email.
+            </p>
+          </FtlMsg>
+        </Banner>
       )}
-      {linkStatus === 'damaged' && <LinkDamaged linkType="reset-password" />}
-      {/* TODO update ResetPasswordLinkExpired to receive sendNewLinkEmail() */}
-      {linkStatus === 'expired' && <LinkExpired linkType="reset-password" />}
-    </>
+      {BannerState.PasswordResetSuccess === bannerState && (
+        <Banner type={BannerType.success}>
+          <FtlMsg id="account-recovery-reset-password-success-alert">
+            <p>Password set</p>
+          </FtlMsg>
+        </Banner>
+      )}
+
+      <CardHeader
+        headingText="Create new password"
+        headingTextFtlId="create-new-password-header"
+      />
+      <FtlMsg id="account-restored-success-message">
+        <p className="text-sm mb-4">
+          You have successfully restored your account using your account
+          recovery key. Create a new password to secure your data, and store it
+          in a safe location.
+        </p>
+      </FtlMsg>
+
+      {/* Hidden email field is to allow Fx password manager
+        to correctly save the updated password. Without it,
+        the password manager tries to save the old password
+        as the username. */}
+      <input type="email" value={state.email} className="hidden" readOnly />
+      <section className="text-start mt-4">
+        <FormPasswordWithBalloons
+          {...{
+            formState,
+            errors,
+            trigger,
+            register,
+            getValues,
+            passwordMatchErrorText,
+            setPasswordMatchErrorText,
+          }}
+          passwordFormType="reset"
+          onSubmit={handleSubmit(
+            (data: FormData) => {
+              onSubmit(data);
+            },
+            (err) => {
+              console.error(err);
+            }
+          )}
+          email={state.email}
+          loading={false}
+          onFocusMetricsEvent={`${viewName}.engage`}
+        />
+      </section>
+
+      <LinkRememberPassword {...state} />
+    </AppLayout>
   );
+
+  /**
+   * Determines starting state for component
+   */
+  function getInitialState() {
+    let email = '';
+    let linkStatus: LinkStatus = LinkStatus.valid;
+    let forceAuth = false;
+    let supportsRecovery = true;
+    let contextError: ContextValidationErrors | null = null;
+
+    try {
+      email = verificationInfo.email || '';
+
+      forceAuth = !!verificationInfo.forceAuth;
+
+      if (!verificationInfo.isValid()) {
```

Comment:
Does this introduce a second way of checking link validity (in addition to the `useLinkStatus` hook that @LZoog created)? Are the use cases different, or should one replace the other?

Reply:
@vpomerleau I'm working on a HOC to kind of combine the two approaches. IMO I like the hooks, but, Dan's actual validation is nicer, and I think we can get the best of both worlds.